tree/homework2.php

```diff
@@ -0,0 +1,84 @@
+<?php
+require_once ('homework1.php');
+
+class Queue
+{
+    private $elements;
+
+    public function __construct()
+    {
+        $this->elements = array();
+    }
+
+    public function enqueue($num) {
+        array_unshift($this->elements, $num);
+    }
+
+    public function dequeue() {
+        if (!$this->isEmpty()) {
+            unset($this->elements[sizeof($this->elements) - 1]);
+        }
+    }
+
+    public function front() {
+        if (!$this->isEmpty()) {
+            return $this->elements[sizeof($this->elements) - 1];
+        }
+        return null;
+    }
+
+    public function isEmpty() {
+        return empty($this->elements);
+    }
+}
+
+function inserting (object $bt, object $insert) {
+    $queue1 = new Queue;
+    if ($bt->getRoot() != null) {
+        $queue1->enqueue($bt->getRoot());
```

Comment:
Moving you `while` loop below this line ;)

Reply:
GATCHA